pkg/git/repository/service.go

```diff
@@ -0,0 +1,27 @@
+package repository
+
+import (
+	"github.com/redhat-developer/git-service/pkg/git"
+)
+
+type Service interface {
+	GetListOfFilesInRootDir() ([]string, error)
+	GetLanguageList() ([]string, error)
+}
+
+type ServiceCreator func(gitSource *git.Source) (Service, error)
+
+func NewService(gitSource *git.Source, serviceCreators []ServiceCreator) (Service, error) {
```

Comment:
This name is a little abstract, could we call it `NewGitService` or something more specific instead?

Reply:
Good point - looks better now?: 19eb36e